pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilterProvider.java

```diff
@@ -37,74 +43,85 @@
 import org.apache.pulsar.common.util.ObjectMapperFactory;
 
 @Slf4j
-public class EntryFilterProvider {
+public class EntryFilterProvider implements AutoCloseable {
 
     @VisibleForTesting
     static final String ENTRY_FILTER_DEFINITION_FILE = "entry_filter";
 
-    /**
-     * create entry filter instance.
-     */
-    public static ImmutableMap<String, EntryFilterWithClassLoader> createEntryFilters(ServiceConfiguration conf,
-                                                                                      EntryFilters entryFilters)
+    private final ServiceConfiguration serviceConfiguration;
+    @VisibleForTesting
+    protected Map<String, EntryFilterMetaData> definitions;
+    @VisibleForTesting
+    protected Map<String, NarClassLoader> cachedClassLoaders;
```

Comment:
is it required that we refer to NarClassLoader here ? can it be simply a "ClassLoader" ?

Reply:
no because we need to get the ServiceDefinition file from the classloader, so a NarClassLoader is required